src/main/java/org/folio/circulation/services/agedtolost/AssignLostFeesWhenAgedToLostService.java

```diff
@@ -0,0 +1,286 @@
+
+    return feeFineRepository.getAutomaticFeeFines(lostItemFeeTypes())
+      .thenApply(r -> r.combine(allLoansResult, this::mapFeeFineTypesToLoans));
+  }
+
+  private List<LoanToAssignFees> mapFeeFineTypesToLoans(Collection<FeeFine> feeTypes,
+     List<LoanToAssignFees> allLoans) {
+
+    return allLoans.stream()
+      .map(loanToAssignFees -> loanToAssignFees.withFeeFineTypes(feeTypes))
+      .collect(Collectors.toList());
+  }
+
+  private CompletableFuture<Result<List<LoanToAssignFees>>> fetchFeeFineOwners(
+    List<LoanToAssignFees> allLoans) {
+
+    final Set<String> uniqueEffectiveLocationServicePoints = allLoans.stream()
+      .map(LoanToAssignFees::getOwnerServicePointId)
+      .collect(Collectors.toSet());
+
+    return feeFineOwnerRepository.findOwnersForServicePoints(uniqueEffectiveLocationServicePoints)
+      .thenApply(r -> r.map(owners -> mapOwnersToLoans(owners, allLoans)));
+  }
+
+  private List<LoanToAssignFees> mapOwnersToLoans(Collection<FeeFineOwner> owners,
+    List<LoanToAssignFees> loans) {
+
+    final Map<String, FeeFineOwner> servicePointToOwner = new HashMap<>();
+
+    owners.forEach(owner -> owner.getServicePoints()
+      .forEach(servicePoint -> servicePointToOwner.put(servicePoint, owner)));
+
+    return loans.stream()
+      .map(loanToAssignFees -> loanToAssignFees.withOwner(servicePointToOwner))
+      .collect(Collectors.toList());
+  }
+
+  private CompletableFuture<Result<MultipleRecords<Loan>>> fetchLoansAndItems() {
+    return loanFetchQuery()
+      .after(query -> loanRepository.findByQuery(query, MAXIMUM_NUMBER_OF_LOANS_TO_PROCESS))
+      .thenComposeAsync(loansResult -> itemRepository.fetchItemsFor(loansResult, Loan::withItem))
+      .thenComposeAsync(r -> r.after(lostItemPolicyRepository::findLostItemPoliciesForLoans));
+  }
+
+  private Result<CqlQuery> loanFetchQuery() {
+    final DateTime currentDate = getClockManager().getDateTime();
+
+    final Result<CqlQuery> billingDateQuery = lessThanOrEqualTo(BILLING_DATE_PROPERTY, currentDate);
+    final Result<CqlQuery> agedToLostQuery = exactMatch(ITEM_STATUS, AGED_TO_LOST.getValue());
+    final Result<CqlQuery> hasNotBeenBilledQuery = exactMatch(
+      LOST_ITEM_HAS_BEEN_BILLED_PROPERTY, "false");
+
+    return Result.combine(billingDateQuery, agedToLostQuery, CqlQuery::and)
+      .combine(hasNotBeenBilledQuery, CqlQuery::and)
+      .map(query -> query.sortBy(ascending(BILLING_DATE_PROPERTY)));
+  }
+
+  private Result<LoanToAssignFees> validateCanCreateAccountForLoan(LoanToAssignFees loanToAssignFees) {
+    if (loanToAssignFees.hasNoFeeFineOwner()) {
+      log.warn("No fee/fine owner present for service point {}, skipping loan {}",
+        loanToAssignFees.getOwnerServicePointId(), loanToAssignFees.getLoan().getId());
+
+      return failed(singleValidationError("No fee/fine owner found for item's effective location",
+        "servicePointId", loanToAssignFees.getOwnerServicePointId()));
```

Comment:
As this is a background process, what happens to this failed result?

Reply:
It should be logged in okapi logs, I think.

Reply:
Meaning that the failure is provided in the response?

Reply:
Yes